### PR TITLE
Add topic diagnostics

### LINKED ISSUE #28: Add diagnostics
Add diagnostics to gather topic statistics for the `image_raw` topic.

### PR DESCRIPTION
**Pull request by icolwell**
Closes #28 

Add a diagnostic updater to monitor publish frequency and data age.

It's been tested once before, but could use one last round of car testing, will add to a test request PR shortly.

### CODE REVIEW

File: src/ximea_ros_cam.cpp [hunk 1]
```diff
@@ -110,6 +114,23 @@ void XimeaROSCam::initNodeHandles() {
     ROS_INFO("... Node Handles Loaded. ");
 }
 
+void XimeaROSCam::initDiagnostics() {
+    if (this->enable_diagnostics) {
+        this->frequency_min = this->cam_framerate_set_ - this->frequency_tolerance;
```
Comment (icolwell) [inline src/ximea_ros_cam.cpp]: Currently I'm using the `this->cam_framerate_set_` parameter, but that may only be valid if `frame_rate_control: true`. How is the frequency regulated in this driver?

Comment (carloswanguw) [reply src/ximea_ros_cam.cpp]: It is regulated based on when the camera has an image in the buffer, so it is entirely camera based.

I plan on adding some actual polling duration for the camera images to decrease some cpu loading

Comment (icolwell) [reply src/ximea_ros_cam.cpp]: ok, I'll just add a parameter for the fps limits, and let the user set them as needed.

**Comment by icolwell** [comment]
Tested today with hardware, works as expected.

**Pull request merged**